core/modules/widgets/radio.js

```diff
@@ -83,9 +82,24 @@ RadioWidget.prototype.setValue = function() {
 };
 
 RadioWidget.prototype.handleChangeEvent = function(event) {
+	var variables = Object.create(null);
 	if(this.inputDomNode.checked) {
 		this.setValue();
 	}
+	// Trigger actions. Use variables = {key:value, key:value ...}
+	if(this.radioActions) {
+		$tw.utils.each(this.attributes,function(val,key) {
+			if(key.charAt(0) !== "$") {
```

Comment:
@Jermolene @pmario So is the intention that all attributes starting with `$` are a reserved name space for future attributes that we might want to add to the widget?  By allowing users to use arbitrary attribute names, in effect every attribute name becomes a reserved one, as it could clash with widget attributes that we introduce in the future.

 At the moment it is just `action-widgets` that have attributes starting with `$`. This is confusing enough as it is but at least users can remember the rule that non-action widget attribute names do not start with `$`. 

It would be quite confusing if suddenly any widget could have attribute names that may or not start with `$`. Especially if the widget previously has many attributes with no $ prefix and then suddenly has one new one introduced with a `$` prefix.

Reply:
What would you suggest?

Reply:
I could imagine, that "user_attributes" start with an underscore. eg: `_user-attribute` and `reserved-for-core` and `$reserved-for-core`

Reply:
or: `__attribute` with 2 underscores

Reply:
@pmario I don't have an obvious solution that I particularly love.

However I think it is **important** that:
a) we try to avoid the potential confusion with attribute names if we can.
b) if we are going to proceed with this change, it needs to be a conscious decision where we consider the benefit of declaring variables in this manner vs the cost of creating usability issues and decide if this change is worthwhile.

In terms of other implementation approaches I considered requiring all user declared attributes to have a specific prefix, like `var-`. The problem is that while that makes it easier to identify user attributes, it makes it trickier to loop through the attributes and identify which is a widget "native" attribute since they have no specific prefix. 

Every widget could have a list of attributes that get turned into variables, plus any user ones with a given prefix.

So in pseudo-code:
```
var widgetAttributes = ["value","name",label"];
$tw.utils.each(this.attributes,function(val,key) {
    if(nativeAttributes.indexOf(key) !== -1 || key.substring(0,4) === "var-"){
        // create a variable for this attribute.
    }
}
```

The other option we could consider is not to enforce a prefix for user attributes in the code, but rather make it a part of documentation and best practice. Explaining that using user attributes without a reserved prefix like "var-" could cause upgrade problems in the future. The problem with this approach is that I suspect most users will ignore it, and we will still have support issues when attribute names do clash in the future.

Does any of this perhaps inspire any ideas for you?

Reply:
@pmario I didn't get to see your comments while typing. But yes, a prefix for user declared attributes would solve the problem.

Reply:
I'm sometimes using a similar approach in my plugins, to make attribute code "loopable", if there are many of them, or if I'm not sure about the names. So it's easy to change the names, because I only need to change the innitial array.

So IMO @Jermolene should have a closer look.

Reply:
@jermolene to try and summarize for you:

- allowing any arbitrary name for attributes assigned by users to create variables, means that every attribute name is potentially a reserved name when/if we want to add more widget attributes in the future

- The code in this PR reserves the `$` prefix namespace for attributes. So presumably future widget attributes would have to start with `$`. This is a usability concern, as outside of action widgets users can reliably expect widget attributes not to start with `$`. Imagine if the `Button` widget which has been around for a long time and has many attributes, suddenly has a new attribute that starts with `$`.

- One suggestion to mitigate this is, to require user assigned widget attributes to have a specific prefix, such as `_` or even `attr-` (in which case the name of the custom attribute and the variable arising from it could be the same).

Reply:
The behaviour here is so novel compared to other widgets that I favour a big obvious prefix on the user defined attributes that are to be passed through to the action string. I think `var-xxx`makes some sense for both the attribute names and variable names because it makes it clearer that this is a mini version of the `<$vars>` widget...